src/v/cloud_topics/frontend/frontend.cc

```diff
@@ -239,15 +335,18 @@ kafka::offset frontend::local_start_offset() const {
 
 kafka::offset frontend::start_offset() const {
     // Ask partition for its start offset
-    // TODO: query metadata layer to get the actual start offset.
-    // the 'partition::sync_kafka_start_offset_override' is not invoked here
-    // because it's tied to both archival_metadata_stm and log_eviction_stm.
-    // For cloud topics we will do log eviction differently and the
-    // DeleteRecords API is not implemented yet. So the code is just
-    // using the start_offset of the Raft log at the moment which is incorrect.
+    auto eviction_stm
+      = _partition->raft()->stm_manager()->get<cluster::log_eviction_stm>();
+    auto override = eviction_stm->kafka_start_offset_override();
+    if (override != kafka::offset{}) {
+        return override;
+    }
```

Comment:
I need to remember to talk through the race condition with L1 having an earlier start offset compared to L0 and how we handle that, because log_eviction_stm only looks to see if the prefix truncate command is before the local log offset.